infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentHandle.java

```diff
@@ -57,19 +58,19 @@ public synchronized void release() {
 
     public void close() {
         Set<FileChannel> channels =
-                new HashSet(Arrays.asList(writeChannel, readChannel, trimmedChannel, pendingTrimChannel));
+                new HashSet<>(Arrays.asList(writeChannel, readChannel, trimmedChannel, pendingTrimChannel));
         for (FileChannel channel : channels) {
             try {
                 channel.force(true);
+            } catch (IOException e) {
+                log.debug("Can't force updates in the channel", e.getMessage());
+            }
+
+            try {
                 channel.close();
-                channel = null;
             } catch (Exception e) {
                 log.warn("Error closing channel {}: {}", channel.toString(), e.toString());
             }
         }
-
-        knownAddresses = null;
```

Comment:
There's a minor change of behavior here: as with making these fields final, the corresponding references are not being released by close() anymore.

Reply:
In addition: https://www.ibm.com/developerworks/java/library/j-jtp01274/index.html#3.2
Especially section: Immutability, again